agent/fw_wordpress.c

```diff
@@ -21,156 +22,73 @@
-    if (!nr_php_is_zval_valid_string(path)) {
-      nrl_verbosedebug(NRL_FRAMEWORK,
-                       "%s: unexpected non-string value in path array",
-                       __func__);
-    } else {
-      if (first) {
-        first = 0;
-      } else {
-        nr_buffer_add(buf, NR_PSTR("|"));
-      }
-
-      nr_regex_add_quoted_to_buffer(buf, Z_STRVAL_P(path),
-                                    zval_len_without_trailing_slash(path));
-    }
-  }
-  ZEND_HASH_FOREACH_END();
-  nr_buffer_add(buf, NR_PSTR(")"));
-  add_wildcard_path_component(buf);
-  nr_buffer_add(buf, NR_PSTR("\0"));
-
-  /*
-   * If the first flag is still set, no valid theme paths existed and we
-   * should abort.
-   */
-  if (first) {
-    nrl_verbosedebug(NRL_FRAMEWORK, "%s: no valid elements in the path array",
-                     __func__);
-  } else {
-    regex = nr_regex_create(nr_buffer_cptr(buf), NR_REGEX_CASELESS, 1);
-  }
-
-  nr_buffer_destroy(&buf);
-  return regex;
-}
-
-static nr_regex_t* compile_regex_for_constant(const char* constant,
-                                              const char* suffix TSRMLS_DC) {
-  nr_regex_t* regex = NULL;
-  zval* value = nr_php_get_constant(constant TSRMLS_CC);
+static nr_matcher_t* create_matcher_for_constant(const char* constant,
+                                                 const char* suffix) {
+  zval* value = nr_php_get_constant(constant);
 
   if (nr_php_is_zval_valid_string(value)) {
-    nrl_verbosedebug(NRL_FRAMEWORK,
-                     "Wordpress: found value = %s for constant=%s",
-                     Z_STRVAL_P(value), constant);
-
-    nrbuf_t* buf = nr_buffer_create(2 * Z_STRLEN_P(value), 0);
-
-    nr_regex_add_quoted_to_buffer(buf, Z_STRVAL_P(value),
-                                  zval_len_without_trailing_slash(value));
-    nr_buffer_add(buf, suffix, nr_strlen(suffix));
-    add_wildcard_path_component(buf);
-    nr_buffer_add(buf, NR_PSTR("\0"));
+    nr_matcher_t* matcher = nr_matcher_create();
+    char* prefix = nr_formatf("%s%s", Z_STRVAL_P(value), suffix);
 
-    regex = nr_regex_create(nr_buffer_cptr(buf), NR_REGEX_CASELESS, 1);
+    nr_matcher_add_prefix(matcher, prefix);
 
-    nr_buffer_destroy(&buf);
-  } else {
-    /*
-     * If the constant isn't set, that's not a problem, but if it is and it's
-     * an unexpected type we should log a message.
-     */
-    if (value) {
-      nrl_verbosedebug(NRL_FRAMEWORK, "%s: unexpected non-string value for %s",
-                       __func__, constant);
-    }
+    nr_free(prefix);
+    nr_php_zval_free(&value);
+    return matcher;
   }
 
   nr_php_zval_free(&value);
-  return regex;
+  return NULL;
 }
 
-static char* try_match_regex(const nr_regex_t* regex, const char* filename) {
-  char* plugin = NULL;
-  nr_regex_substrings_t* ss = NULL;
-
-  ss = nr_regex_match_capture(regex, filename, nr_strlen(filename));
-  if (NULL == ss) {
+static char* nr_wordpress_strip_php_suffix(char* filename) {
```

Comment:
Seems like `nr_path_basename` is a better name? Maybe make it a utility function?

Reply:
See comment here: https://github.com/newrelic/newrelic-php-agent/pull/765#discussion_r1408464349

Reply:
This comment is outdated.